packages/koa-shopify-webhooks/src/register.ts

```diff
@@ -41,21 +48,46 @@ export async function registerWebhook({
 
   const result = await response.json();
 
-  if (
-    result.data &&
-    result.data.webhookSubscriptionCreate &&
-    result.data.webhookSubscriptionCreate.webhookSubscription
-  ) {
-    return {success: true, result};
-  } else {
-    return {success: false, result};
+  return {success: isSuccess(result, deliveryMethod), result};
+}
+
+function isSuccess(result, deliveryMethod: DeliveryMethod): boolean {
+  switch (deliveryMethod) {
+    case DeliveryMethod.Http:
+      return Boolean(
+        result.data &&
+          result.data.webhookSubscriptionCreate &&
+          result.data.webhookSubscriptionCreate.webhookSubscription,
+      );
+    case DeliveryMethod.EventBridge:
+      return Boolean(
+        result.data &&
+          result.data.eventBridgeWebhookSubscriptionCreate &&
+          result.data.eventBridgeWebhookSubscriptionCreate.webhookSubscription,
+      );
   }
```

Comment:
Minor: Seems we can use [optional chaining](https://stackoverflow.com/questions/40238144/typescript-the-safe-navigation-operator-or-and-null-property-paths) to simplify this a bit if you're interested:
```suggestion
  switch (deliveryMethod) {
    case DeliveryMethod.Http:
      return Boolean(result.data?.webhookSubscriptionCreate?.webhookSubscription);
    case DeliveryMethod.EventBridge:
      return Boolean(result.data?.eventBridgeWebhookSubscriptionCreate?.webhookSubscription);
  }
```
What you have so far is also perfectly fine and valid.

Reply:
Yeah that's awesome, a lot cleaner!

Reply:
The tests are giving a syntax error... we're probably behind on the versions of some things. Going to revert to the older code :/

Reply:
Darn. Well it was worth a shot, thanks for trying!